rootston/wl_shell.c

```diff
@@ -57,11 +57,8 @@ static void handle_destroy(struct wl_listener *listener, void *data) {
 	struct roots_wl_shell_surface *roots_surface =
 		wl_container_of(listener, roots_surface, destroy);
 	wl_list_remove(&roots_surface->destroy.link);
-	wl_list_remove(&roots_surface->ping_timeout.link);
```

Comment:
Why did these go? You need to remove the listener link to stop listening to the event.

Reply:
Because `ping_timeout` and the others are not used, there's no listener. See the header: https://github.com/swaywm/wlroots/pull/303/files#diff-fe0ba5038842e9d235dfd2047ed4d031L28